libtsuba/src/RDG.cpp

```diff
@@ -693,6 +693,28 @@ UnloadProperty(
   return KATANA_CHECKED(props->RemoveColumn(i));
 }
 
+katana::Result<katana::Uri>
+GetStorageLocationIfValid(
+    const std::string& name,
+    const std::vector<katana::PropStorageInfo>& prop_info_list) {
+  auto psi_it = std::find_if(
+      prop_info_list.begin(), prop_info_list.end(),
+      [&](const katana::PropStorageInfo& psi) { return psi.name() == name; });
+  if (psi_it == prop_info_list.end()) {
+    return KATANA_ERROR(
+        katana::ErrorCode::PropertyNotFound, "no property named {}",
+        std::quoted(name));
+  }
+
+  if (!(psi_it->IsAbsent() || psi_it->IsClean())) {
+    return KATANA_ERROR(
+        katana::ErrorCode::AssertionFailed, "the property exists but is dirty");
+  }
+  // TODO(thunt) there's really no reason why we shouldn't always use uri
+  auto path = KATANA_CHECKED(katana::Uri::Make(psi_it->path()));
```

Comment:
My only objection in this code is the use of the term `path`.  The function returns a file name, not a path.  I thought by path, it would be the path name including rdg_dir.  But it isn't.